File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--designing-an-outlook-form-region_2.vb
    Private Sub MapIt_FormRegionShowing(ByVal sender As System.Object, _
        ByVal e As System.EventArgs) Handles MyBase.FormRegionShowing
        Dim tempLoc As String = ""
        Dim defaultAddress As String = ""
        Dim scratchPadAddress As String = ""

        Dim myItem As Outlook.ContactItem = _
            CType(Me.OutlookItem, Outlook.ContactItem)

        If Not (myItem Is Nothing) Then
            If Not (myItem.HomeAddress Is Nothing) And _
                myItem.HomeAddress.Trim().Length > 0 Then
                tempLoc = myItem.HomeAddressStreet.Trim() + " " _
                    + myItem.HomeAddressCity + " " + myItem.HomeAddressState + _
                        " " + myItem.HomeAddressPostalCode
                If myItem.HomeAddress = myItem.MailingAddress Then
                    defaultAddress = tempLoc + "_Home"
                Else
                    scratchPadAddress += "adr." + tempLoc + "_Home~"
                End If
            End If
            If Not (myItem.BusinessAddress Is Nothing) And _
                myItem.BusinessAddress.Trim().Length > 0 Then
                tempLoc = myItem.BusinessAddressStreet.Trim() + " " _
                    + myItem.BusinessAddressCity + " " + _
                        myItem.BusinessAddressState + " " + _
                            myItem.BusinessAddressPostalCode
                If myItem.BusinessAddress = myItem.MailingAddress Then
                    defaultAddress = tempLoc + "_Business"
                Else
                    scratchPadAddress += "adr." + tempLoc + "_Business~"
                End If
            End If
            If Not (myItem.OtherAddress Is Nothing) And _
                myItem.OtherAddress.Trim().Length > 0 Then
                tempLoc = myItem.OtherAddressStreet.Trim() + " " + _
                    myItem.OtherAddressCity + " " + myItem.OtherAddressState + _
                        " " + myItem.OtherAddressPostalCode
                If myItem.OtherAddress = myItem.MailingAddress Then
                    defaultAddress = tempLoc + "_Other"
                Else
                    scratchPadAddress += "adr." + tempLoc + "_Other~"
                End If
            End If
        End If

        WebBrowser1.Navigate(("http://local.live.com/default.aspx?style=r&where1=" _
            + defaultAddress + "&sp=" + scratchPadAddress))

    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--designing-an-outlook-form-region_2.cs
        private void MapIt_FormRegionShowing(object sender, EventArgs e)
        {
            string tempLoc = "";
            string defaultAddress = "";
            string scratchPadAddress = "";

            Outlook.ContactItem myItem = (Outlook.ContactItem)this.OutlookItem;

            if (myItem != null)
            {
                if (myItem.HomeAddress != null && 
                        myItem.HomeAddress.Trim().Length > 0)
                {
                    tempLoc = myItem.HomeAddressStreet.Trim() + " " + 
                        myItem.HomeAddressCity + " " + myItem.HomeAddressState + 
                            " " + myItem.HomeAddressPostalCode;
                    if (myItem.HomeAddress == myItem.MailingAddress)
                    {
                        defaultAddress = tempLoc + "_Home";
                    }
                    else
                    {
                        scratchPadAddress += "adr." + tempLoc + "_Home~";
                    }
                }
                if (myItem.BusinessAddress != null && 
                        myItem.BusinessAddress.Trim().Length > 0)
                {
                    tempLoc = myItem.BusinessAddressStreet.Trim() + 
                        " " + myItem.BusinessAddressCity + " " + 
                            myItem.BusinessAddressState + " " + 
                                myItem.BusinessAddressPostalCode;
                    if (myItem.BusinessAddress == myItem.MailingAddress)
                    {
                        defaultAddress = tempLoc + "_Business";
                    }
                    else
                    {
                        scratchPadAddress += "adr." + tempLoc + "_Business~";
                    }
                }
                if (myItem.OtherAddress != null && myItem.OtherAddress.Trim().Length > 0)
                {
                    tempLoc = myItem.OtherAddressStreet.Trim() + " " + 
                        myItem.OtherAddressCity + " " + myItem.OtherAddressState + 
                            " " + myItem.OtherAddressPostalCode;
                    if (myItem.OtherAddress == myItem.MailingAddress)
                    {
                        defaultAddress = tempLoc + "_Other";
                    }
                    else
                    {
                        scratchPadAddress += "adr." + tempLoc + "_Other~";
                    }
                }
            }

            webBrowser1.Navigate("http://local.live.com/default.aspx?style=r&where1=" 
                + defaultAddress + "&sp=" + scratchPadAddress);

        }